app/decorators/allocation_decorator.rb

```diff
@@ -10,4 +10,42 @@ def hourly_rate
   def to_s
     "##{model.id}"
   end
+
+  def user_name
+    model.user.first_and_last_name
+  end
+
+  def project_name
+    model.project ? model.project.name : nil
```

Comment:
Instead, you could use delegate in the model. https://api.rubyonrails.org/classes/Module.html#method-i-delegate